File: Java & Python/linked_lists/remove_duplicates_sorted_linked_list_ii.py
# if A.val == A.next.val do another loop to see if it matches the current duplicate
class Solution:
	# @param A : head node of linked list
	# @return the head node in the linked list
	def deleteDuplicates(self, A):
		head = result = ListNode('dummy')

		while A:
			if A.next and A.val == A.next.val:
				duplicate = A.val
				while A and A.val == duplicate:
					A = A.next
			else:
				head.next = A
				head = head.next
				A = A.next
		head.next = None
		return result.next
"""
Remove Duplicates from Sorted List IIBookmark
Given a sorted linked list, delete all nodes that have duplicate numbers, leaving only distinct numbers from the original list.

For example,
Given 1->2->3->3->4->4->5, return 1->2->5.
Given 1->1->1->2->3, return 2->3.
"""

File: Java & Python/linked_lists/remove_duplicates_sorted_linked_list_ii.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     public int val;
 *     public ListNode next;
 *     ListNode(int x) { val = x; next = null; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode root) {
    	
    	ListNode head = new ListNode(0);
    	ListNode result = head;

		while (root != null) {
			if (root.next != null && root.val == root.next.val) {
				int duplicate = root.val;
				while (root != null && root.val == duplicate)
					root = root.next;
			}
			else {
				head.next = root;
				head = head.next;
				root = root.next;
			}
		}
		head.next = null;
		return result.next;
    }
}